datasetIngestor/checkMetadata_test.go

```diff
@@ -0,0 +1,118 @@
+package datasetIngestor
+
+import (
+	"net/http"
+	"testing"
+	"time"
+	"reflect"
+)
+
+func TestGetHost(t *testing.T) {
+    // Call the function under test.
+		host := GetHost()
+
+    // fail the test and report an error if the returned hostname is an empty string.
+    if len(host) == 0 {
+        t.Errorf("getHost() returned an empty string")
+    }
+
+    // fail the test and report an error if the returned hostname is "unknown".
+    if host == "unknown" {
+        t.Errorf("getHost() was unable to get the hostname")
+    }
+}
+
+func TestCheckMetadata(t *testing.T) {
+	// Define mock parameters for the function
+	var TEST_API_SERVER string = "https://dacat-qa.psi.ch/api/v3" // "https://example.com/api"
```

Comment:
I think we shouldn't connect to dacat for real, but rather fake the call response. If we do this, this test has no dependency on dacat and will work even if dacat will change address or if other RIs want to use the CLI

Reply:
The problem is the tests fail with not information if I change the server.

Reply:
could it be that it fails [here](https://github.com/paulscherrerinstitute/scicat-cli/blob/3c7a584def4f406a306a2981611d18c8f081d127/datasetIngestor/checkMetadata.go#L228)? If so, and if there isn't a way to mock it in go, I think we could proceed with this PR and do a refactor after that, using in line 228 the client (which we can mock)

Reply:
maybe in PR #17

Reply:
> could it be that it fails [here](https://github.com/paulscherrerinstitute/scicat-cli/blob/3c7a584def4f406a306a2981611d18c8f081d127/datasetIngestor/checkMetadata.go#L228)?

I tested that too. I created local urls for all the cases `myurl` would be. It fails without any error message.

But yeah, we either can spend an unknown amount of time here, or just mark this part of the test to be improved in the future. Additionally, the APIServer is not an important part of the `CheckMetadata` function, it is used in `GetProposal`, for which we will have independent tests.

#17 just makes the code cleaner and more robust, the logic of the code remains the same.

Shall I mark this line as a TODO?

Reply:
I think a TODO is fine, maybe you could open an issue for that, so we don't forget. I am fine as well to do it outside of #17, good point.

Reply:
Ok, I'll create an issue after this is merged.